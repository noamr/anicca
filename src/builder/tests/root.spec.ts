/// </ reference "jest" />
import nearley from 'nearley'
const grammar = require('../root.ne')

const parse = (str: string) => new nearley.Parser(grammar).feed(str).finish()[0]

describe('Root', () => {
    it('rules', () => {
        expect(parse('Controller abc')).toMatchSnapshot()
        expect(parse('View abc')).toMatchSnapshot()
        expect(parse('Main')).toMatchSnapshot()
        expect(parse('Slot abcSlot')).toMatchSnapshot()
        expect(parse('Bus bus')).toMatchSnapshot()
    })
})