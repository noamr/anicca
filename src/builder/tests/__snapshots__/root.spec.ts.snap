// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root rules 1`] = `
Object {
  "name": "abc",
  "type": "Controller",
}
`;

exports[`Root rules 2`] = `
Object {
  "name": "abc",
  "type": "View",
}
`;

exports[`Root rules 3`] = `
Object {
  "type": "Main",
}
`;

exports[`Root rules 4`] = `
Object {
  "name": "abcSlot",
  "type": "Slot",
}
`;

exports[`Root rules 5`] = `
Object {
  "name": "bus",
  "type": "Bus",
}
`;
